TestHelpers.Tests/ConvertersTests.cs

```diff
@@ -0,0 +1,54 @@
+using System.Collections;
+using System.Collections.Generic;
+using NUnit.Framework;
+
+namespace System.IO.Abstractions.TestingHelpers.Tests
+{
+    [TestFixture]
+    public class ConvertersTests
```

Comment:
Alright, I think I understand everything that's going on here, so this is my opinion.

To me, since the Converters are internal to the System.IO.Abstractions, and the consumers of our library will never directly interact with the Converters. I don't think we should directly test them. 

I think we should add lazy tests, but they should be against the public interface, EnumerateFiles(). That way we're calling it the same way our end users would be. Just because our Converters work and are lazy, doesn't mean the public interface always will be. Someone could technically come along and gut the EnumerateFiles() method and replace the implementation with something else. Our tests would still pass, but the method would no longer be lazy.

Thoughts?

Reply:
Hm, I agree with your thoughts regarding testing private methods, but I have a different opinion for internal classes. In my opinion the "unit" of a unit test should be somehow self-contained and testable in isolation and does not not have to be the whole API. 

Testing the lazyness at the API level for me would bring unnecessary complexity to the tests without any clear value.

Regarding "Just because our Converters work and are lazy, doesn't mean the public interface always will be" and "Someone could technically come along and gut the EnumerateFiles() method and replace the implementation with something else. Our tests would still pass, but the method would no longer be lazy.": To mitigate exactly these kinds of problems you have the concept of integration tests. Those verify that a bunch of (unit tested) parts are composed in the right way.

Reply:
I don't think it would necessarily add complexity. We would have a test that simply calls EnumerateFiles with a test name of something like EnumerateFiles_ByDefault_IsLazy(), and that would validate that the method truley is lazy. 

We have a product gap right now where EnumerateFiles is not lazy. Tests prove that a gap/bug has been fixed, but we don't have a test for EnumerateFiles being lazy. We've added code to support it, but we're not proving the method is now lazy. I think thats my biggest catchup right now is that we're not proving what we've done.

Regardless, it's just a matter of opinion so if you're OK with it, then I am. 👍

Reply:
Yeah, that's one of the very few places where we do more than just forwarding calls to `System.IO`. So maybe not worth putting too much time in it. I'll merge this as is now - we might revisit any time later :)